### PR TITLE
Complete first version of word counter

### PR DESCRIPTION
**Pull request by luuthehien1992**

### CODE REVIEW

File: Week-02/word_counter.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		}
+	}
+}
+
+func wordCounter(wg *sync.WaitGroup, files chan string, words chan string) {
+	// Do until files chanel close
+	for file := range files {
+		// Not check error here since if an error occurred that will return when call getNextWord
+		f, _ := os.Open(file)
+		defer f.Close()
```
Comment (havinhdat) [inline Week-02/word_counter.go:48]: please don't ignore this error, if file name is incorrect, will cause nil pointer exception

File: Week-02/word_counter.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	var wordMap = make(map[string]uint64)
+
+	for i = 0; i < MAX_POOL; i++ {
+		wordCounterWG.Add(1)
+		go wordCounter(&wordCounterWG, files, words)
+	}
+
+	go resultCounter(words, wordMap)
+
+	findFiles(path, files)
```
Comment (havinhdat) [inline Week-02/word_counter.go:108]: we can use concurrency here
Channels are used for goroutine cross-communication, if it's not the case, then no need for channel

File: Week-02/word_counter.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+	wordCounterWG.Wait()
+
+	fmt.Println(wordMap)
+}
```
Comment (havinhdat) [inline Week-02/word_counter.go:113]: Thank you for submitting and happy coding 👏

File: Week-02/word_counter.go (hunk 1 above)
Comment (luuthehien1992) [reply Week-02/word_counter.go:48]: I got it.

File: Week-02/word_counter.go (hunk 3 above)
Comment (luuthehien1992) [reply Week-02/word_counter.go:113]: Thanks

File: Week-02/word_counter.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+		case c == ' ':
+			if len(stringBuilder) > 0 {
+				return string(stringBuilder), nil
+			}
+		}
+	}
+}
+
+func wordCounter(wg *sync.WaitGroup, files chan string, words chan string) {
+	// Do until files chanel close
```
Comment (hieuvo) [inline Week-02/word_counter.go:44]: the best practice is, just to make sure you don't forget it at the end of this method
```
defer wg.Done()
```

File: Week-02/word_counter.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+		f, _ := os.Open(file)
+		defer f.Close()
+
+		for {
+			word, err := getNextWord(f)
+
+			if err == io.EOF {
+				break
+			} else if err != nil {
+				log.Fatal(err)
```
Comment (hieuvo) [inline Week-02/word_counter.go:56]: crashing the app because of invalid word doesn't make sense

**Pull request merged**